### PR TITLE
LE AUdio: Fix CSIS client conn refs

### PR DESCRIPTION
**Pull request by Thalley**
Improves how the CSIS client shell handles the connection points and member references.

### CODE REVIEW

File: subsys/bluetooth/shell/csis_client.c [hunk 1]
```diff
@@ -76,8 +68,27 @@ static void connected_cb(struct bt_conn *conn, uint8_t err)
...
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Disconnected: %s (reason 0x%02x)\n", addr, reason);
+
+	for (uint8_t i = 0; i < members_found; i++) {
+		if (set_members[i].conn == conn) {
+			bt_conn_unref(set_members[i].conn);
+			set_members[i].conn = NULL;
+			members_found--;
```
Comment (kruithofa) [inline subsys/bluetooth/shell/csis_client.c:83]: I assume you want to loop until all the members have been found, but here you decrease 'members_found', so you may leave the loop too early

Comment (Thalley) [reply subsys/bluetooth/shell/csis_client.c:83]: A `set_member` represents a remote device with a CSIS. There is a 1:1 relation between a `set_member` and an `bt_conn`, so there should only ever be a single `set_member` where `set_members[i].conn == conn`

Comment (kruithofa) [reply subsys/bluetooth/shell/csis_client.c:83]: Thanks for the explanation; I see what's happening now

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by kruithofa — APPROVED** [review]

**Comment by Thalley** [comment]
Closing as this has been fixed by https://github.com/zephyrproject-rtos/zephyr/pull/47384 in a different way

**Pull request closed without merging**